src/clients/SpokePoolClient/EVMSpokePoolClient.ts

```diff
@@ -210,12 +228,71 @@ export class EVMSpokePoolClient extends SpokePoolClient {
       at: "SpokePoolClient#findDeposit",
       message: "Located deposit outside of SpokePoolClient's search range",
       deposit,
-      elapsedMs: tStop - tStart,
+      elapsedMs,
     });
 
     return { found: true, deposit };
   }
 
+  public override async findAllDeposits(depositId: BigNumber): Promise<MultipleDepositSearchResult> {
+    // First check memory for deposits
+    let deposits = this.getDepositsForDepositId(depositId);
+    if (deposits.length > 0) {
+      return { found: true, deposits };
+    }
+
+    // If no deposits found in memory, try to find on-chain
+    const result = await this.queryDepositEvents(depositId);
+    if ("reason" in result) {
+      return { found: false, code: InvalidFill.DepositIdNotFound, reason: result.reason };
+    }
+
+    const { events, elapsedMs } = result;
+
+    if (events.length === 0) {
+      return {
+        found: false,
+        code: InvalidFill.DepositIdNotFound,
+        reason: `${getNetworkName(this.chainId)} depositId ${depositId} not found at block ${result.from}.`,
+      };
+    }
+
+    // First do all synchronous operations
+    deposits = events.map((event) => {
+      const deposit = {
+        ...spreadEventWithBlockNumber(event),
+        originChainId: this.chainId,
+        fromLiteChain: true,
+        toLiteChain: true,
+      } as DepositWithBlock;
+
+      if (isZeroAddress(deposit.outputToken)) {
+        deposit.outputToken = this.getDestinationTokenForDeposit(deposit);
+      }
```

Comment:
fwiw I think this needs to be re-worked following the Address class updates that were merged on Friday.